remote-keypair/src/openpgp_card.rs

```diff
@@ -0,0 +1,359 @@
+    ///   * AID must be a 16-byte hexadecimal string (32 digits).
+    ///   * D276000124 => AID preamble, fixed across all AIDs
+    ///   * 01         => application ID, fixed to 01 == OpenPGP
+    ///   * 0304       => OpenPGP version, 3.4 in this case
+    ///   * 0006       => unique manufacturer ID, Yubico in this case
+    ///   * 12345678   => smart card serial number
+    ///   * 0000       => reserved for future use
+    ///
+    /// No other URI formats are valid.
+    fn try_from(uri: &URIReference<'_>) -> Result<Self, Self::Error> {
+        let scheme = uri.scheme().map(|s| s.as_str().to_ascii_lowercase());
+        let ident = uri.host().map(|h| h.to_string());
+
+        match (scheme, ident) {
+            (Some(scheme), Some(ident)) if scheme == "pgpcard" => {
+                if ident.is_empty() {
+                    return Ok(Self { aid: None });
+                }
+                if ident.len() != 32 {
+                    return Err(LocatorError::IdentifierParseError(
+                        "OpenPGP AID must be 32 digits".to_string(),
+                    ));
+                }
+                let mut ident_bytes = Vec::<u8>::new();
+                for hex_byte in &ident.chars().chunks(2) {
+                    ident_bytes.push(
+                        u8::from_str_radix(hex_byte.collect::<String>().as_str(), 16).map_err(
+                            |_| {
+                                LocatorError::IdentifierParseError(
+                                    "non-hex character found in identifier".to_string(),
+                                )
+                            },
+                        )?,
+                    );
+                }
+                Ok(Self {
+                    aid: Some(ident_bytes.as_slice().try_into().map_err(|_| {
+                        LocatorError::IdentifierParseError("invalid identifier format".to_string())
+                    })?),
+                })
+            }
+            (Some(scheme), None) if scheme == "pgpcard" => Ok(Self { aid: None }),
+            _ => Err(LocatorError::MismatchedScheme),
+        }
+    }
+}
+
+/// An ongoing connection to the OpenPGP application on a smart card.
+///
+/// Field `pgp: OpenPgp` is a long-lived card access object. When OpenpgpCard
+/// is initialized via TryFrom<&Locator>, this object contains a `PcscBackend`
+/// connector object that communicates with the card using PC/SC.
+///
+/// Actual operations on the card are done through short-lived
+/// OpenPgpTransaction objects that are instantiated from `OpenPgp` on an ad
+/// hoc basis. There should only exist one active OpenPgpTransaction at a time.
+///
+/// Field `pin_verified` indicates whether a successful PIN verification has
+/// already happened in the past (specifically for PW1 used to authorize
+/// signing). This allows redundant PIN verification to be skipped for cards
+/// that require only a single successful PIN entry per session.
+pub struct OpenpgpCard {
```

Comment:
[openpgp_card](https://docs.rs/openpgp-card/0.3.4/openpgp_card/index.html) library uses [`OpenPgp`](https://docs.rs/openpgp-card/0.3.4/openpgp_card/struct.OpenPgp.html) as a capitalization format for the OpenPGP name.  It matches a common convention when an uppercase acronym is written as a separate word in a camel case scheme.
Maybe call this `OpenPgpCard` as well?